### PR TITLE
fix(web-components): fix side navigation 'failed to execute getComputedStyle' console error

### LINKED ISSUE #2795: Side Navigation: Failed to execute 'getComputedStyle'
## Summary of the bug 
While communicating with the customer on Internal Issue 2905 (#2610) they raised another error that was occurring on their example sidenav code.

> Uncaught TypeError: Failed to execute 'getComputedStyle' on 'Window': parameter 1 is not of type 'Element'.

It looks like it's an issue inside our code (it's happening in the `paddingIconWidth` function in ic-side-navigation.tsx. I get the exact same error when using your example code in Stackblitz, and there's a similar 'style' error reported in our [React Storybooks](https://mi6.github.io/ic-ui-kit/branches/develop/react/?path=/story/react-components-side-navigation--react-router-with-collapsed-icon-and-label) Note: this is inconsistent* recommend using the below stackblitz.

*there is a chance that the fix for #2610 has solved this bug. The error was about icon styles and the initial issue was with icons failing to appear, now that they appear consistently it might now be fixed?

## 🪜 How to reproduce 
Tell us the steps to reproduce the problem: 
1. Run this [Stackblitz provided by the user](https://stackblitz.com/edit/rc77dn-bytkex?file=src%2Fapp.tsx)
2. Click on: '....' 
3. Type in: '....' 
4. See the error 

If the Stackblitz link doesn't work copy this code in:
```js
import React, { useState } from "react";
import { IcSideNavigation, IcNavigationItem } from "@ukic/react";
import "@ukic/react/dist/core/core.css";
import "@ukic/react/dist/core/normalize.css";

const menus = {
  mainMenu: [
    {
      name: "Home",
      link: "/",
    },
    {
      name: "Search",
      link: "/search",
    },
    {
      name: "News",
      menu: "news",
    },
    {
      name: "My Account",
      menu: "account",
    },
    {
      name: "Settings",
      link: "/settings",
    },
  ],
  news: [
    {
      name: "Back",
      menu: "mainMenu",
    },
    {
      name: "Local",
      link: "/local-news",
    },
    {
      name: "Politics",
      link: "politics-news",
    },
    {
      name: "Sports",
      link: "/sports",
    },
  ],
  account: [
    {
      name: "Back",
      menu: "mainMenu",
    },
    {
      name: "Preferences",
      link: "/preferences",
    },
    {
      name: "User Settings",
      link: "/user-settings",
    },
  ],
};

const App = () => {
  const [menu, setMenu] = useState("mainMenu");
  const currentMenu = menus[menu];

  const items = currentMenu.map((item) => {
    const onMenuClick = () => {
      setMenu(item.menu);
    };
    const onLinkClick = () => {
      console.log("Clicked link");
    };

    const onClick = item.link
      ? onLinkClick
      : item.menu
        ? onMenuClick
        : () => {};
    const expandable = !!item.menu && item.name !== "Back";

    return (
      <IcNavigationItem
        label={item.name}
        onClick={onClick}
        slot="primary-navigation"
        expandable={expandable}
      />
    );
  });
  return (
    <IcSideNavigation appTitle="My Website" expanded>
      {items}
    </IcSideNavigation>
  );
};

export default App;
```

## Additional info 
please refer to the internal issue mentioned above to contact the internal customer.

### PR DESCRIPTION
**Pull request by lz405**
## Summary of the changes
added null handling if there is no icon for navigation item

## Related issue
fix #2795 

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 
- [x] Compare performance of modified components against develop using Performance addon in React Storybook.

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.
- [x] Props/slots can be updated after initial render.

### CODE REVIEW

File: packages/web-components/src/components/ic-side-navigation/ic-side-navigation.tsx [hunk 1]
```diff
@@ -424,7 +424,7 @@ export class SideNavigation {
 
     const navStyles = {
       gap: window.getComputedStyle(navItemLink).gap,
-      iconWidth: window.getComputedStyle(navItemSVG).width,
+      iconWidth: navItemSVG ? window.getComputedStyle(navItemSVG).width : "0",
```
Comment (evenstensberg) [inline packages/web-components/src/components/ic-side-navigation/ic-side-navigation.tsx:427]: Maybe add an optional chaining to the method object and add it to the rest of the methods that call the same function? 

Best would be to add the check to the function itself. This way you guarantee any calls to the function will return something anywhere

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Comment (evenstensberg) [inline packages/web-components/src/components/ic-side-navigation/ic-side-navigation.tsx:427]:
```suggestion
      iconWidth: navItemSVG ? window.getComputedStyle(navItemSVG)?.width : "0",
```

**Review by GCHQ-Developer-299 — CHANGES_REQUESTED** [review]
While clicking rapidly between React sidenav stories (how i've made the error occur before) I found that style errors do still occur sometimes in another method - screenshot attached:
Screenshot 2024-11-25 at 09 45 27

**Comment by evenstensberg** [comment]
@GCHQ-Developer-299 Thats why you should either use optional changing and ternaries, or null-ish coacending in all your util methods.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Comment by gd2910** [comment]
LGTM

**Pull request merged**